diff -uprN a/bootable/recovery/Android.mk b/bootable/recovery/Android.mk
--- a/bootable/recovery/Android.mk	2024-02-02 12:48:29.600442691 +0800
+++ b/bootable/recovery/Android.mk	2024-02-02 12:53:05.844329099 +0800
@@ -78,6 +78,11 @@ LOCAL_SRC_FILES := \
     twrpAdbBuFifo.cpp \
     twrpRepacker.cpp

+# Debug
+LOCAL_CFLAGS += -g
+LOCAL_CPPFLAGS += -g
+LOCAL_STRIP_MODULE := false
+
 ifneq ($(TARGET_RECOVERY_REBOOT_SRC),)
   LOCAL_SRC_FILES += $(TARGET_RECOVERY_REBOOT_SRC)
 endif
